opentracing/harness/api_check.py

```diff
@@ -88,10 +89,11 @@ def mock_finish(*_):
         else:
             raise AssertionError('Expected ValueError')  # pragma: no cover
 
-    def test_span_tags(self):
+    def test_span_tags_with_chaining(self):
         span = self.tracer().start_trace(operation_name='Farnsworth')
-        span.add_tag('birthday', '9 April, 2841')
-        span.add_tag('loves', 'different lengths of wires')
+        span. \
+            set_tag('birthday', '9 April, 2841'). \
+            set_tag('loves', 'different lengths of wires')
         span.finish()
```

Comment:
well, that's the problem with a "black hole" nature of the Span API, we can write to it but we can't read back. The only assertion here is that (a) the span supports the `set_tag` method with the right number of arguments (actually I should've used named parameters for even tighter binding, although key/value are trivial), and (b) the method supports chaining.

As I mentioned elsewhere, one way to actually test reading is to define a special "introspection" interface that a tracer can implement with a special form of reporter, so that after writing data to the span we can assert that it's written by calling this introspection API. But even that isn't great given that tracers are allowed to drop tags on the floor.